CONTRIBUTING.md

```diff
@@ -1,31 +1,368 @@
-clear and has sufficient instructions to be able to reproduce the issue.
 
-Facebook has a [bounty program](https://www.facebook.com/whitehat/) for the safe
-disclosure of security bugs. In those cases, please go through the process
-outlined on that page and do not file a public issue.
+# License
+
+By contributing to the tutorials, you agree that your contributions will
+be licensed as described in the `LICENSE` file in the root directory of
+this source tree.
+
+
+# Updates to existing tutorials
+
+We welcome your pull requests (PR) for updates and fixes.
+
+1. If you haven't already, complete the Contributor License Agreement
+   ("CLA").
+1. Fork the repo and create a branch from
+   [`master`](https://github.com/pytorch/tutorials).
+1. Test your code.
+1. Lint your code with a tool such as
+   [Pylint](https://pylint.pycqa.org/en/latest/).
+1. Submit your PR for review.
+
+
+# New Tutorials
+
+There are three types of tutorial content that we host on
+[`pytorch.org/tutorials`](https://github.com/pytorch/tutorials):
+
+* **Interactive tutorials** are authored and submitted as Python files.
+  The build system  converts these into Jupyter notebooks and HTML. The
+  code in these tutorials is run every time they are built. To keep
+  these tutorials up and running all their package dependencies need to
+  be resolved--which makes it more challenging to maintain this type of
+  tutorial. 
+
+* **Non-interactive tutorials** are authored and submitted as
+  reStructuredText files. The build system only converts them into HTML;
+  the code in them does not run on build. These tutorials are easier to
+  create and maintain but they do not provide an interactive experience.
+  An example is the [Dynamic Quantization
+  tutorial](https://pytorch.org/tutorials/recipes/recipes/dynamic_quantization.html).
+
+* **Recipes** are tutorials that provide bite-sized, actionable
+  examples of how to use specific features, which differentiates them
+  from full-length tutorials. Recipes can be interactive or
+  non-interactive.
+
+
+# Managing data that is used by your tutorial 
+
+Your tutorial might depend on external data, such as pre-trained models,
+training data, or test data. We recommend storing this data in a
+commonly-used storage service, such as Amazon S3, and instructing your
+users to download the data at the beginning of your tutorial. 
+
+The
+[Makefile](https://github.com/pytorch/tutorials/blob/master/Makefile)
+that we use to build the tutorials contains automation that downloads
+required data files.
+
+
+# Python packages used by your tutorial
+
+We recommend that you use only mature, well-supported packages in your
+tutorial. Packages that are obscure or not well-maintained may break as
+a result of, for example, updates to Python or PyTorch or other
+packages. If your tutorial fails to build in our Continuous Integration
+(CI) system, we might contact you in order to resolve the issue.
+
```

Comment:
We can probably say that something like: If your tutorial relies on dependancies not defined in the requirements.txt file, you can add them there.

Reply:
I've added this text to the beginning of this section, so that they add any extra packages they need to `requirements.txt`.